src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -138,6 +138,9 @@ private sealed class RunningInstance
             // Callback delegate for the EventStream events
             private readonly Interop.EventStream.FSEventStreamCallback _callback;
```

Comment:
The new GCHandle is a conterpart for this delegate. The RunningInstance has to be kept alive for as long as the marshalled delegate is in use. There was nothing to guarantee that before this change.

I have also looked at into replacing the delegate by a function pointer, but that was more involved change so I have abandoned it.